target/linux/ath79/dts/qca9558_belkin_f9j1108-v2.dts

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca955x.dtsi"
+
+/ {
+	model = "Belkin F9J1108v2 (AC1750 DB Wi-Fi)";
+	compatible = "belkin,f9j1108-v2", "qca,qca9558";
+
+	aliases {
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		label-mac-device = &eth1;
+	};
+	
+	leds {
+		compatible = "gpio-leds";
+		
+		usb {
+			label = "f9j1108-v2:blue:usb2";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port1>, <&hub_port0>;
+			linux,default-trigger = "usbport";
+		};
+		
+		wps-amber {
+			label = "f9j1108-v2:amber:wps";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+		
+		wps-blue {
+			label = "f9j1108-v2:blue:wps";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+		};
+
+		led_system: system {
+			label = "f9j1108-v2:blue:system";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+	};
+	
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio_usb_power {
+			gpio-export,name = "f9j1108-v2:usb2:power";
+			gpio-export,output = <1>;
+			gpios = <&gpio 21 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+		
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie1 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&usb_phy0 {
+	status = "okay";
+};
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port0: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&usb_phy1 {
+	status = "okay";
+};
+
+&usb1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port1: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		mtdparts: partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+			
+			partition@50000 {
+				compatible = "openwrt,okli";
+				label = "firmware";
+				reg = <0x050000 0xe20000>;
+			};
+			
+			partition@e70000 {
+				label = "loader";
+				reg = <0xe70000 0x180000>;
```

Comment:
if you want to reclaim most of this space you can use the following partitions here:
```
			fwpart1: partition@50000 {
				label = "fwpart1";
				reg = <0x050000 0xe20000>;
			};
			partition@e70000 {
				label = "loader";
				reg = <0xe70000 0x10000>;
				read-only;
			};
			fwpart2: partition@e80000 {
				label = "fwpart2";
				reg = <0xe80000 0x150000>;
			};
```
then in the same section as the keys or LEDs:
```
	virtual_flash {
		compatible = "mtd-concat";

		devices = <&fwpart1 &fwpart2>;

		partitions {
			compatible = "fixed-partitions";
			#address-cells = <1>;
			#size-cells = <1>;

			partition@0 {
				compatible = "openwrt,okli";
				label = "firmware";
				reg = <0x0 0x0>;
			};
		};
	};
```

Reply:
Thanks, I was wondering how to do that, and copy/paste worked perfectly:

/dev/mtdblock9           10304       420      **9884**   4% /overlay